### PR TITLE
[FIX] web: use language week_start in datepickers

### LINKED ISSUE #36450: [12.0][BUG] Datepicker start-of-week is not affected by locale editing
1. Edit English locale in Odoo to change week_start to Monday (let's leave out reasoning for it)
2. Odoo calendar now starts on Monday, but datepicker still starts on Sunday, since it uses the unmodified locale settings.

Possible reason:
https://github.com/odoo/odoo/blob/4b8e0d82bb2463e9d5e3492f7014c89cdc41b5e7/addons/web/static/src/js/views/calendar/calendar_renderer.js#L370

Shouldn't here be a call to `datepickerLocale` https://github.com/odoo/odoo/blob/905e01921f3c3ef43ace6fc537d1d8c0d280002c/addons/web/static/lib/fullcalendar/js/fullcalendar.js#L10360 as well to configure the locale of datepicker with custom settings of Odoo?

### PR DESCRIPTION
**Pull request by andriisem**
Before this rev., if the week_start param (the first day of week) of the user language
was changed, it wasn't reflected on datepickers (however, it worked fine in the
calendar view). This rev. makes this work by updating the moment locale with the
corresponding param.

Fixes #36450
Closes #36532

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by qsm-odoo** [comment]
@aab-odoo Maybe something to think about ? But there are probably more parameters to use in the updated locale...

**Comment by aab-odoo** [comment]
duplicate of #36532

File: addons/web/static/src/js/core/session.js [hunk 1]
```diff
@@ -253,6 +254,14 @@ var Session = core.Class.extend(mixins.EventDispatcherMixin, {
         });
         return this.qweb_mutex.def;
     },
+    configure_locale: function () {
+        moment.updateLocale(moment.locale(), {
+            week: {
+                dow: (_t.database.parameters.week_start || 0) % 7,
+            },
+        });
+        return Promise.resolve();
```
Comment (aab-odoo) [inline addons/web/static/src/js/core/session.js]: you can remove this line

File: addons/web/static/src/js/core/session.js [hunk 2]
```diff
@@ -253,6 +254,14 @@ var Session = core.Class.extend(mixins.EventDispatcherMixin, {
         });
         return this.qweb_mutex.def;
     },
+    configure_locale: function () {
```
Comment (aab-odoo) [inline addons/web/static/src/js/core/session.js]:
```
    /**
     * Sets first day of week in current locale according to the user language.
     *
     * @private
     */
    _configureLocale: function () {
        ...
    },
```

**Comment by andriisem** [comment]
Hello, @aab-odoo, thanks for review. I updated your remarks

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**